backend/backup_media.js

```diff
@@ -81,16 +81,37 @@ const saveEntry = (mediaType, outputPath, outputFilename, mediaSrc, itemType, it
   fs.writeFileSync(config.lastProcessedIdFile, itemId.toString());
 };
 
-// replaces part of the url with asterisks
-//  https://i.imgur.com/7IElab.jpeg =>  https://i.imgur.com/7IE***.jpeg
 const hideUrl = (url) => {
-    let parts = url.split('/');
-    let lastPart = parts[parts.length - 1];
-    let lastPartParts = lastPart.split('.');
-    lastPartParts[0] = lastPartParts[0].substr(0, 3) + '***';
-    lastPart = lastPartParts.join('.');
-    parts[parts.length - 1] = lastPart;
-    return parts.join('/');
+    try {
+        // parse url
+        const parsedUrl = new URL(url);
+
+        // replace second half of the characters in the string with asterisks
+        const hideChars = (str) => {
```

Comment:
minor: I would define this in the global scope similar to other functions, up to you

Reply:
I usually prefer to reduce the namespace scope as much as possible, and performance is not a concern here.